react_juce/core/ViewManager.cpp

```diff
@@ -87,11 +87,47 @@ namespace reactjuce
     {
         const auto& [view, shadow] = getViewHandle(viewId);
 
-        // ShadowView::setProperty returns true when a layout prop
-        // has been set.  Otherwise set on the view and repaint
-        if(!shadow->setProperty(name, value)) {
-          view->setProperty(name, value);
-          view->repaint();
+        auto parseStyleObj = [=] (const juce::var &obj)
+        {
+            const StyleSheetID id = obj;
+            auto it = styleSheets.find(id);
+
+            if (it != styleSheets.end())
+            {
+                juce::NamedValueSet &style = it->second;
+
+                for (auto &nv : style)
+                {
+                    //TODO: We'll likely change this to some internal
+                    //      setStyleProperty on View which isn't directly
+                    //      exposed to our react renderer implemenation.
+                    setViewProperty(viewId, nv.name.toString(), nv.value);
```

Comment:
@nick-thompson, so here is the main thing to decide on in this MR I think. To get things moving and test the overall approach I simply unpack the stored stylesheet props and call `setViewProperty` recursively. 

We don't want to support the old style props on `View` anymore and instead only want to support passing `styles` from React. So, how should we approach this?

Ideas, 

We have a `View::setStyleProperty` function which we call into here. `View::setStyleProperty` will only ever be called internally and never by the renderer/React. This would work fine in theory, then in `View::setProperty` we only handle the properties we expect/accept. We could also combine this with your earlier idea of filtering out unsupported properties in `Backend.ts`'s `setProperty`. 

Any alternative ideas/thoughts. Hopefully we can pair this up with actually typing all supported `ViewProps` etc. I'm happy to do that as a separate MR or just pack onto the top of this one.

Reply:
Yea good question.

So, for one, I don't think we need to worry about any recursive behavior here. It's fine for style properties to deal with nested objects, like our `border` conversation. The user won't work with a nested object, but our macros will expand out into a border object. I think probably the only reason we might need recursive application here is just because thats how our current view painting expects it. But, we should just change that.

Second, I actually don't think the View even needs to hold these properties. It's fine to just hold a single `style` property in the `View` itself, then at paint time lookup the ViewManager and read the relevant styles from the ViewManager's map. That will prevent potentially a lot of value copying.

And then third, yea, I think filtering (and logging a warning message) at the js layer is definitely high on the priority list.

That make sense? This can also work out such that we can merge this without introducing a hgue breaking change: in our `View` paint routine, we check: do I have a `style` property? If so, lookup my styles from the ViewManager, and paint with them, then return early. If not, fall back to the current paint routine to support those properties (and perhaps log a deprecation warning). Then in a future PR we can remove support for the current way of doing things. It's a bit unclear to me how careful we have to be with backwards compat... I suspect we're still within a window of time where a breaking change like this won't affect too many people too drastically, so maybe I'm overthinking

Reply:
Riiiight ok. Approach makes total sense. So are we going with supporting the "deprecated" approach for now? Or do we make the breaking change? Think we have issue for filtering props. That should be able to be dealt with as a follow up to this I think. I'm making this ticket, full prop types and then props filtering my priorities (in that order).

Reply:
> I'm making this ticket, full prop types and then props filtering my priorities (in that order).

Sounds good!

> So are we going with supporting the "deprecated" approach for now? Or do we make the breaking change?

Let's do the easy version of deprecating: 
* In View::paint, if we have a style prop we look up our style props via the ViewManager and paint according to those, then return
* Else, we DBG("Deprecation warning: painting from top-level properties"); and then carry on with the code that's already written there

In a future PR we'll do prop filtering at the js level which will make the latter branch obsolete and we can remove it then

Reply:
@nick-thompson , just coming back to this. So quick question, if we have View holding a single `style` prop which back references to the style object map in `ViewManager` how do you envisage we should set `ShadowView` properties ? At the moment we have the below approach for this:

```
            if(!shadow->setProperty(name, value)) {
              view->setProperty(name, value);
              view->repaint();
            }
```

Reply:
Yea, I think they would need to. We could try to differentiate between layout props and others, but I think it's probably fine for both the view and its shadow to have a style prop

Reply:
Yep cool. I'll take that approach then. OK cheers. Try to get this finished off tomorrow.

Reply:
Just thinking about how we should give `View` access to the `ViewManager` to fetch a stylesheet object. Right now I can see two options, 

1. We add a getter for `ViewManager` to `ReactApplication` root and fetch it that way. 

2. We go for composition and have some sort of `StyleSheetFactory` or similar which `ViewManager` owns and then we pass an instance of `StyleSheetFactory` into `ViewFactory` when creating `View` instances so that it can be passed to `View`'s constructor. We might need to make that an overload used mostly internally to avoid the change to `ViewFactory` being a breaking change if possible. 

What do you think?

Reply:
One thought. I don't think we can avoid `ViewFactory` taking a `StyleSheetFactory` being a breaking change if we do go down that route. It's probably the nicest route I think but not sure how you feel about this one.